pandas/core/groupby/groupby.py

```diff
@@ -5829,6 +5808,63 @@ def func(df):
                     stacklevel=find_stack_level(),
                 )
 
+        if axis == 1:
+            try:
+                if self.obj.ndim == 1:
+                    result = self._op_via_apply(how, skipna=skipna)
+                else:
+
+                    def func(df):
+                        method = getattr(df, how)
+                        return method(
+                            axis=axis, skipna=skipna, numeric_only=numeric_only
+                        )
+
+                    func.__name__ = how
+                    result = self._python_apply_general(
+                        func, self._obj_with_exclusions, not_indexed_same=True
+                    )
+            except ValueError as err:
+                name = "argmax" if how == "idxmax" else "argmin"
+                if f"attempt to get {name} of an empty sequence" in str(err):
+                    raise ValueError(
+                        f"Can't get {how} of an empty group due to unobserved "
+                        "categories. Specify observed=True in groupby instead."
+                    ) from None
```

Comment:
from None?

Reply:
https://docs.python.org/3/library/exceptions.html#exception-context

> The expression following [from](https://docs.python.org/3/reference/simple_stmts.html#raise) must be an exception or None. It will be set as __cause__ on the raised exception. Setting __cause__ also implicitly sets the __suppress_context__ attribute to True, so that using raise new_exc from None effectively replaces the old exception with the new one for display purposes (e.g. converting [KeyError](https://docs.python.org/3/library/exceptions.html#KeyError) to [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError)), while leaving the old exception available in __context__ for introspection when debugging.

Reply:
We want to completely overwrite the prior traceback here, rather than presenting both to the user.